static/app/views/performance/transactionSummary/transactionOverview/eapChartsWidget.tsx

```diff
@@ -71,13 +75,22 @@ type EAPChartsWidgetProps = {
 };
 
 export function EAPChartsWidget({transactionName}: EAPChartsWidgetProps) {
-  const [selectedWidget, setSelectedWidget] = useState<EAPWidgetType>(
-    EAPWidgetType.DURATION_BREAKDOWN
-  );
   const location = useLocation();
-  const spanCategoryUrlParam = decodeScalar(
-    location.query?.[SpanIndexedField.SPAN_CATEGORY]
-  );
+  const navigate = useNavigate();
+
+  const params = useLocationQuery({
+    fields: {
+      [SpanIndexedField.SPAN_CATEGORY]: decodeScalar,
+      [SELECTED_CHART_QUERY_PARAM]: decodeScalar,
+    },
+  });
+
+  const spanCategoryUrlParam = params[SpanIndexedField.SPAN_CATEGORY];
+  const {chartDisplay: selectedChartUrlParam} = params;
```

Comment:
Why not
```suggestion
  const spanCategoryUrlParam = params[SpanIndexedField.SPAN_CATEGORY];
  const {chartDisplay: selectedChartUrlParam, [SpanIndexedField.SPAN_CATEGORY]: spanCategoryUrlParam} = params;
```

or even just put that up on like 81 where you `useLocationQuery`?

Reply:
True, I'll clean things up here